testplan/testing/multitest/entries/assertions.py

```diff
@@ -155,6 +155,46 @@ class Equal(FuncAssertion):
     label = "=="
     func = operator.eq
 
+    def __init__(self, first, second, description=None, category=None):
+        self.highlight_expected = {}
+        self.highlight_actual = {}
+        self.type_actual = type(first).__name__
+        self.type_expected = type(second).__name__
+        super(Equal, self).__init__(
+            first=first,
+            second=second,
+            description=description,
+            category=category,
+        )
+
+    def evaluate(self):
+        if isinstance(self.first, str) and isinstance(self.second, str):
+            common_chrs_cnt = 0
+            for index in range(min(len(self.first), len(self.second))):
+                if self.first[index] == self.second[index]:
+                    common_chrs_cnt += 1
+                else:
+                    break
+            if common_chrs_cnt > 0:
+                self.highlight_actual = {
+                    "eq": (0, common_chrs_cnt),
+                    "ne": (common_chrs_cnt, len(self.first)),
+                }
+                self.highlight_expected = {
+                    "eq": (0, common_chrs_cnt),
+                    "ne": (common_chrs_cnt, len(self.second)),
+                }
+            else:
+                self.highlight_actual = {
+                    "ne": (common_chrs_cnt, len(self.first))
+                }
+                self.highlight_expected = {
```

Comment:
This here creating a set instead of a dictionary.

Since this logic is always highlighting from 0 to b and b to end it is enough to store the common char count.

Though I feel this might be just a display details, and might not need to be in t he assertion at all.

Reply:
Let's just save the "b" in the assertion?